### PR TITLE
Docker images: support both linux and windows

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -21,12 +21,53 @@ jobs:
...
+      - name: Log in to Azure Container Registry
+        run: |
+          echo ${{ secrets.ACR_PASSWORD }} | docker login ${{ secrets.ACR_LOGIN_SERVER }} -u ${{ secrets.ACR_USERNAME }} --password-stdin
+
+      - name: Create Multi-Platform Image
+        run: |
+          docker buildx imagetools create --tag ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }} \
+            ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-windows \
+            ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-linux
+          docker buildx imagetools create --tag ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:latest \
```
Comment (squirrelsc) [inline .github/workflows/docker-image.yml:66]: Does it need to pull the tags from remote, or it's all done by local? If it needs to pull from remote, this should be just tag, then push.

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -21,12 +21,53 @@ jobs:
...
-          docker push ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}
-          docker push ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:latest
+          docker build -t ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-linux -f Dockerfile.linux .
+          docker push ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-linux
+
+      - name: Log out from ACR
+        run: |
+          docker logout ${{ secrets.ACR_LOGIN_SERVER }}
+
+  build-windows:
```
Comment (squirrelsc) [inline .github/workflows/docker-image.yml:10]: To reduce the network traffic, how about below steps:
1. (windows task) build windows image, and push to remote.
2. (linux) Let it depend on first step. 
    1. build linux image
    2. pull windows image
    3. buildx the bundle

File: Dockerfile.linux [hunk 3]
```diff
@@ -29,4 +29,3 @@ WORKDIR /app/lisa
 
 RUN python3 -m pip install --no-cache-dir --upgrade pip && \
     python3 -m pip install --no-cache-dir --editable .[azure,libvirt,baremetal] --config-settings editable_mode=compat
```
Comment (squirrelsc) [inline Dockerfile.linux]: Add ado

Comment (LiliDeng) [reply Dockerfile.linux]: > Add ado

I don't understand this.

Comment (squirrelsc) [reply Dockerfile.linux]: Include the extra “ado” like azure, lobbies. All dev extra needs to be added.

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -21,12 +43,44 @@ jobs:
...
-      - name: Build and Push Docker Image
+      - name: Build and Push Linux Docker Image
         run: |
-          docker build -t ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }} .
-          docker tag ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }} ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:latest
-          docker push ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}
-          docker push ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:latest
+          docker build -t ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-linux -f Dockerfile.linux .
+          docker push ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-linux
+          docker push ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:latest-linux
```
Comment (squirrelsc) [inline .github/workflows/docker-image.yml]: Does it need to push latest linux?

**Comment by LiliDeng** [comment]
execution flow

image

File: .github/workflows/docker-image.yml [hunk 5]
```diff
@@ -7,26 +7,70 @@ on:
...
         run: |
           echo ${{ secrets.ACR_PASSWORD }} | docker login ${{ secrets.ACR_LOGIN_SERVER }} -u ${{ secrets.ACR_USERNAME }} --password-stdin
 
-      - name: Build and Push Docker Image
+      - name: Create Multi-Platforms Image
+        run: |
+          docker buildx imagetools create --tag ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }} \
+            ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-windows \
+            ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:${{ github.event.release.tag_name }}-linux
+          docker buildx imagetools create --tag ${{ secrets.ACR_LOGIN_SERVER }}/${{ secrets.ACR_REPOSITORY }}:latest \
```
Comment (squirrelsc) [inline .github/workflows/docker-image.yml:66]: How long does this step take? Can it just tag the previous one, instead of buildx?

**Review by squirrelsc — DISMISSED** [review]

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**